operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -368,17 +368,25 @@ private void validateCRDWithLocalModelIfRequired(Class<P> resClass, String contr
     }
   }
 
-  public void changeNamespaces(Set<String> namespaces) {
+  public synchronized void changeNamespaces(Set<String> namespaces) {
     if (namespaces.contains(WATCH_CURRENT_NAMESPACE)) {
       throw new OperatorException("Unexpected value in target namespaces: " + namespaces);
     }
     if (namespaces.contains(Constants.WATCH_ALL_NAMESPACES) && namespaces.size() > 1) {
       throw new OperatorException(
           "Watching all namespaces, but additional specific namespace is present");
     }
-    eventProcessor.stop();
+    // if the processor was not running, for example because the controller
+    // was not leading in a HA setup, we don't want to stop and
+    // mainly start the processor on namespace change.
+    boolean eventProcessorWasRunning = eventProcessor.isRunning();
+    if (eventProcessorWasRunning) {
+      eventProcessor.stop();
+    }
     eventSourceManager.changeNamespaces(namespaces);
-    eventProcessor.start();
+    if (eventProcessorWasRunning) {
+      eventProcessor.start();
+    }
```

Comment:
could it happen that at this point we are are already the leader (and accidentally not start processing) or the `synchronised` will take care of that

Reply:
exactly, `synchronized` takes care of that